### ISSUE TITLE
Add QueueFile#remove(int) and FileObjectQueue#remove(int)

### ISSUE THREAD

**Issue by f2prateek** [square/tape#66]
Proposing this as the counterpart to #61, which lets us read an arbitrary number of items from QueueFile. Once those elements are processed, we could remove those elements all at once from the queue

```java
List<Foo> list = fileObjectQueue.peek(max);
process(list);
fileObjectQueue.remove(list.size());
```

**Comment by pforhan** [comment]
Possibly reasonable, though this is pretty simple to implement in calling code in the meantime.

```java
    try {
      for (int i = 0; i < count; i++) {
        queue.remove();
      }

    } catch (Exception ex) {
      ...
    }
```

 We really need to get working on the 2.0 of the api...

**Comment by JakeWharton** [comment]
@pforhan that is not atomic

**Comment by f2prateek** [comment]
Closed with #68
